### PR TITLE
User management

### PR DESCRIPTION
**Pull request by ppelszynski**

### CODE REVIEW

File: spec/models/school_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe School, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (mateuszwu) [inline spec/models/school_spec.rb]: You don't have tests here, so you can remove this file ✂️ 

EDIT: 
I can't add comment to `spec/support/features_helper.rb` file because it's empty. Remove it too :)

File: spec/features/sign_in_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+feature 'User sign in', type: :feature do
+  scenario 'correctly login user' do
+    create(:user)
+    visit new_user_session_path
+
+    fill_in 'Email', with: 'user@example.com'
+    fill_in 'Password', with: 'password'
+    select "Public", :from => "statusSelect"
```
Comment (mateuszwu) [inline spec/features/sign_in_spec.rb]: Do we need this?

File: spec/features/school_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+feature 'User can manage schools', type: :feature do
```
Comment (mateuszwu) [inline spec/features/school_spec.rb]: We'll discuss it later :]

File: spec/factories/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user do
+    email { 'user@example.com' }
```
Comment (mateuszwu) [inline spec/factories/user.rb]: Please add `sequence` 🔢

File: spec/factories/schools.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :school do
+    name { 'Example School' }
```
Comment (mateuszwu) [inline spec/factories/schools.rb]: Please check if you need those brackets in attributes with simple value like: string, bool, number

File: db/migrate/20190619134855_rename_user_id_in_schools.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class RenameUserIdInSchools < ActiveRecord::Migration[5.2]
+  def change
+    rename_column :schools, :user_id, :admin_id
```
Comment (mateuszwu) [inline db/migrate/20190619134855_rename_user_id_in_schools.rb]: until it's merge into development / master i recommend you to update `db/migrate/20190619131949_add_user_id_to_schools.rb` migration :)

File: db/migrate/20190619132408_change_closed_default_in_school.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class ChangeClosedDefaultInSchool < ActiveRecord::Migration[5.2]
+  def change
+    change_column_default :schools, :closed, false
```
Comment (mateuszwu) [inline db/migrate/20190619132408_change_closed_default_in_school.rb]: until it's merge into development / master i recommend you to update `db/migrate/20190619114036_create_schools.rb` migration :)

File: db/migrate/20190619131949_add_user_id_to_schools.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToSchools < ActiveRecord::Migration[5.2]
+  def change
+    add_column :schools, :user_id, :integer
```
Comment (mateuszwu) [inline db/migrate/20190619131949_add_user_id_to_schools.rb]: Why `add_column` not `add_reference`?

File: db/migrate/20190618164039_devise_create_users.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class DeviseCreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :users do |t|
```
Comment (mateuszwu) [inline db/migrate/20190618164039_devise_create_users.rb]: I would add here: `first_name`, `last_name`, `phone_number`

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :schools
+
+  # Include default devise modules. Others available are:
```
Comment (mateuszwu) [inline app/models/user.rb]: Remove comments ✂️

File: app/models/school.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+class School < ApplicationRecord
+  belongs_to :admin, class_name: 'User', foreign_key: 'user_id'
+
+  # belongs_to :user
```
Comment (mateuszwu) [inline app/models/school.rb]: ✂️

File: Gemfile [hunk 12]
```diff
@@ -42,19 +44,23 @@ gem 'bootsnap', '~> 1.4.4', require: false
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   gem 'rspec-rails'
+
+  gem 'pry-rails'
```
Comment (mateuszwu) [inline Gemfile]: I think that it should be in separate commit

File: Gemfile [hunk 13]
```diff
@@ -7,6 +7,8 @@ gem 'rubocop', require: false
 
 gem 'capybara'
 
+gem 'factory_bot_rails'
```
Comment (mateuszwu) [inline Gemfile]: I think that it should be in separate commit

File: .rubocop.yml [hunk 14]
```diff
@@ -0,0 +1,38 @@
+AllCops:
```
Comment (mateuszwu) [inline .rubocop.yml]: I think that it should be in separate commit

File: .gitignore [hunk 15]
```diff
@@ -25,4 +25,6 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
-/config/database.yml
\ No newline at end of file
+/config/database.yml
```
Comment (mateuszwu) [inline .gitignore]: I think that it should be in separate commit

File: spec/features/sign_in_spec.rb (hunk 2 above)
Comment (ppelszynski) [reply spec/features/sign_in_spec.rb]: I don't really know why it's there. It's related to schools. My bad.
